jupyterhub/proxy.py

```diff
@@ -366,7 +379,11 @@ async def check_routes(self, user_dict, service_dict, routes=None):
             for name, spawner in user.spawners.items():
                 if spawner.ready:
                     spec = spawner.proxy_spec
+                    spec2 = url_path_join('/socket/', spawner.proxy_spec)
+                    spec3 = url_path_join(spawner.proxy_spec, '/logout/')
```

Comment:
same here: what about /hub/logout? here you even store the URLs in variables -> make them getters and use them above as well